src/Map.php

```diff
@@ -794,14 +794,27 @@ public function offsetUnset($key)
     // implements \IteratorAggregate
 
     /**
-     * Get an iterator for a copy of the map.
+     * Get an iterator for the map.
      *
-     * @return \ArrayIterator
+     * @return \Traversable
      * @api
      */
     public function getIterator()
     {
-        return new \ArrayIterator($this->toArray());
+        return (function () {
+            $keys  = $this->keys();
+            $count = count($keys);
+            $index = 0;
+
+            while ($index < $count) {
+                $key   = $keys[$index];
+                $value = $this->get($key);
+
+                yield $key => $value;
+
+                $index++;
+            }
+        })();
```

Comment:
Why the indirection through an anonymous function?

Reply:
Ah yeah there's probably no need for it.